sqlmesh/dbt/target.py

```diff
@@ -167,6 +175,14 @@ class SnowflakeConfig(TargetConfig):
     retry_on_database_errors: bool = False
     retry_all: bool = False
 
+    @root_validator(pre=True)
+    def validate_authentication(
+        cls, values: t.Dict[str, t.Union[t.Tuple[str, ...], t.Optional[str], t.Dict[str, t.Any]]]
+    ) -> t.Dict[str, t.Union[t.Tuple[str, ...], t.Optional[str], t.Dict[str, t.Any]]]:
+        if values.get("password") or values.get("authenticator"):
```

Comment:
Should we make sure that both `password` and `user` are present, in case `authenticator` isn't used?

Reply:
Note that this is similar to connection.py:

```python
    @root_validator(pre=True)
    def _validate_authenticator(
        cls, values: t.Dict[str, t.Optional[str]]
    ) -> t.Dict[str, t.Optional[str]]:
        if "type" in values and values["type"] != "snowflake":
            return values
        auth = values.get("authenticator")
        user = values.get("user")
        password = values.get("password")
        if not auth and (not user or not password):
            raise ConfigError("User and password must be provided if using default authentication")
        return values
```

Reply:
`user` is a required field in `dbt`. If not set, pydantic will throw an error.

Reply:
Gotcha, thanks for clarifying.